virttest/env_process.py

```diff
@@ -560,6 +560,25 @@ def preprocess(test, params, env):
     :param env: The environment (a dict-like object).
     """
     error_context.context("preprocessing")
+
+    # For KVM to work in Power8 systems we need to have SMT=off
+    # and it needs to be done as root, here we do a check whether
+    # we satisfy that condition, if not try to make it off
+    # otherwise throw TestError with respective error message
+    cmd = "grep cpu /proc/cpuinfo | awk '{print $3}' | head -n 1"
+    cpu_output = avocado_process.system_output(cmd, shell=True).strip().upper()
+    if "POWER8" in cpu_output:
+        utils_misc.verify_running_as_root()
+        smt_output = avocado_process.system_output("ppc64_cpu --smt",
+                                                   shell=True).strip()
+        if smt_output != "SMT is off":
+            try:
+                avocado_process.run("ppc64_cpu --smt=off", verbose=True,
+                                    shell=True)
+                logging.debug("SMT turned off successfully")
+            except Exception, info:
```

Comment:
Better to catch avocado_process.CmdError exception here. And VM will not start because test environment mismatch, so I think give user a clear information "Unable to disable SMT on your Power8 CPU" looks more better than print exception info in here.